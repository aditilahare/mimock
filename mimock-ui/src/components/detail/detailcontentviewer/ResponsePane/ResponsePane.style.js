import tw from 'tailwind-styled-components';

export const NavTabPane = tw.div`
    grid
    grid-cols-1
    gap-6
    fade
    px-4
    py-2
    my-2
`;

export const ContentItem = tw.div`
    mb-3
`;

export const ItemLabel = tw.label`
    form-label
    inline-block
    mb-2
    text-gray-700
`;

export const Item = tw.input`
    form-control
    block
    w-full
    px-3
    py-1.5
    text-base
    font-normal
    text-gray-700
    bg-gray-100 bg-clip-padding
    border border-solid border-gray-300
    rounded
    transition
    ease-in-out
    m-0
    focus:outline-none
`;

export const ItemPreFormat = tw.pre`
    
`;

export const Code = tw.code`
    language-json
`;
