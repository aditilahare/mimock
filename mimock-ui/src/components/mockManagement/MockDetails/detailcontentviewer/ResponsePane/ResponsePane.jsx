import React, { useEffect } from 'react';
import {
	NavTabPane,
	ContentItem,
	ItemLabel,
	Item,
	ItemPreFormat,
	Code,
} from './ResponsePane.style';
import PropTypes from 'prop-types';
import Prism from 'prismjs';

function ResponsePane({ responseHeader, contentType, responseBody }) {
	useEffect(() => {
		Prism.highlightAll();
	}, []);

	return (
		<NavTabPane data-testid='response-pane-section'>
			<ContentItem>
				<ItemLabel>Response Header</ItemLabel>
				<ItemPreFormat>
					<Code>{responseHeader}</Code>
				</ItemPreFormat>
			</ContentItem>
			<ContentItem>
				<ItemLabel>Response Content Type</ItemLabel>
				<Item type='text' defaultValue={contentType} disabled readOnly />
			</ContentItem>
			<ContentItem>
				<ItemLabel>Expected Response</ItemLabel>
				<ItemPreFormat>
					<Code>{responseBody}</Code>
				</ItemPreFormat>
			</ContentItem>
		</NavTabPane>
	);
}

ResponsePane.propTypes = {
	responseHeader: PropTypes.string.isRequired,
	contentType: PropTypes.string.isRequired,
	responseBody: PropTypes.string.isRequired,
};

export default ResponsePane;
