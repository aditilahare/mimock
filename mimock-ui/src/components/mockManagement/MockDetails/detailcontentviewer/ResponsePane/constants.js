export const constants = {
	testIds: {
		responseNavTabPane: 'response-nav-tab-pane',
		responseHeader: 'response-header',
		responseContentType: 'response-content-type',
		expectedResponse: 'expected-response',
	},
	labels: {
		responseHeader: 'Response Header',
		responseContentType: 'Response Content Type',
		expectedResponse: 'Expected Response',
	},
	defaultFieldType: 'text',
};
