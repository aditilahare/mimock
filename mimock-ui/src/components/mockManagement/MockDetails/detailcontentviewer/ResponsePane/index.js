import ResponsePane from './ResponsePane';

export default ResponsePane;
