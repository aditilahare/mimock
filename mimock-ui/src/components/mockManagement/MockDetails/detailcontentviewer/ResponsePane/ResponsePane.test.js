import React from 'react';
import { render } from '@testing-library/react';
import ResponsePane from './ResponsePane';
import { BrowserRouter } from 'react-router-dom';

describe('ResponsePane', () => {
	it('should render response pane component', async () => {
		const tree = await render(<ResponsePane />, { wrapper: BrowserRouter });

		const { container, getByTestId } = tree;

		expect(getByTestId('response-nav-tab-pane')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
