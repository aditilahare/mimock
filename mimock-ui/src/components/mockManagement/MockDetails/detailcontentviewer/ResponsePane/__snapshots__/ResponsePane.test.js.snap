// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsePane should render response pane component 1`] = `
<div>
  <div
    class="grid grid-cols-1 gap-6 fade px-4 py-2 my-2"
    data-testid="response-pane-section"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Response Header
      </label>
      <pre
        class="language-json"
        tabindex="0"
      >
        <code
          class="language-json"
        />
      </pre>
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Response Content Type
      </label>
      <input
        class="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-gray-100 bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:outline-none"
        disabled=""
        readonly=""
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label inline-block mb-2 text-gray-700"
      >
        Expected Response
      </label>
      <pre
        class="language-json"
        tabindex="0"
      >
        <code
          class="language-json"
        />
      </pre>
    </div>
  </div>
</div>
`;
